include/git2/mailmap.h

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (C) the libgit2 contributors. All rights reserved.
+ *
+ * This file is part of libgit2, distributed under the GNU GPL v2 with
+ * a Linking Exception. For full terms see the included COPYING file.
+ */
+#ifndef INCLUDE_git_mailmap_h__
+#define INCLUDE_git_mailmap_h__
+
+#include "common.h"
+#include "types.h"
+
+/**
+ * @file git2/mailmap.h
+ * @brief Mailmap parsing routines
+ * @defgroup git_mailmap Git mailmap routines
+ * @ingroup Git
+ * @{
+ */
+GIT_BEGIN_DECL
+
+/**
+ * A single entry parsed from a mailmap.
+ */
+typedef struct git_mailmap_entry {
+	unsigned int version;
+
+	const char *real_name; /**< the real name (may be NULL) */
+	const char *real_email; /**< the real email (may be NULL) */
+	const char *replace_name; /**< the name to replace (may be NULL) */
+	const char *replace_email; /**< the email to replace */
+} git_mailmap_entry;
+
+#define GIT_MAILMAP_ENTRY_VERSION 1
+#define GIT_MAILMAP_ENTRY_INIT {GIT_MAILMAP_ENTRY_VERSION}
+
+/**
+ * Create a mailmap object by parsing a mailmap file.
+ *
+ * The mailmap must be freed with 'git_mailmap_free'.
+ *
+ * @param out pointer to store the mailmap
+ * @param data raw data buffer to parse
+ * @param size size of the raw data buffer
+ * @return 0 on success
+ */
+GIT_EXTERN(int) git_mailmap_parse(
```

Comment:
To be consistent with `git_mailmap_from_repo`, a function name `git_mailmap_from_buffer` might in fact be favorable. This is only a proposal, the name is fine as is.